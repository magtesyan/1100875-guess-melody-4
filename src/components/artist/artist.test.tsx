import Artist from "./artist";
import * as renderer from "react-test-renderer";
import * as React from "react";
import {GameType, QuestionArtist} from "../../types";

const AVATAR_URL = `https://api.adorable.io/avatars/128`;

const answer = [
  {
    picture: `${AVATAR_URL}/11`,
    artist: `John Snow`,
  }, {
    picture: `${AVATAR_URL}/12`,
    artist: `Jack Daniels`,
  }, {
    picture: `${AVATAR_URL}/13`,
    artist: `Jim Beam`,
  }
];

const question: QuestionArtist = {
  type: GameType.ARTIST,
  song: {
    artist: `Jim Beam`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [{
    picture: `${AVATAR_URL}/11`,
    artist: `John Snow`,
  }, {
    picture: `${AVATAR_URL}/12`,
    artist: `Jack Daniels`,
  }, {
    picture: `${AVATAR_URL}/13`,
    artist: `Jim Beam`,
  }],
};

it(`ArtistQuestionScreen is rendered correctly`, () => {
  const tree = renderer
    .create(<Artist
      index = {1}
      key = {2}
      answer = {answer[0]}
      question = {question}
      onAnswer = {jest.fn()}
    />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
