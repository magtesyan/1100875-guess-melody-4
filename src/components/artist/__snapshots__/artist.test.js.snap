// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen is rendered correctly 1`] = `
<div
  className="artist"
>
  <input
    className="artist__input visually-hidden"
    id="answer-1"
    name="answer"
    onChange={[Function]}
    type="radio"
    value="artist-1"
  />
  <label
    className="artist__name"
    htmlFor="answer-1"
  >
    <img
      alt="John Snow"
      className="artist__picture"
      src="https://api.adorable.io/avatars/128/11"
    />
    John Snow
  </label>
</div>
`;
